client/src/hooks/use-fetch-data.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery, UseQueryOptions } from "@tanstack/react-query";
+import { AxiosError } from "axios";
+
+import api from "@/lib/api";
+
+/**
+ * The configuration object used by the `useFetchData` hook to make API requests.
+ * It extends `UseQueryOptions` from `react-query`, omitting the `queryKey` and `queryFn` properties
+ * since they are specified separately in `useFetchData`.
+ *
+ * @template TData The type of the data being fetched.
+ * @template TError The type of the error, default is `AxiosError`.
+ * @property {any[]} queryKey The unique key for the query, used for caching and refetching.
+ * @property {string} endpoint The endpoint to make the API request to.
+ * @property {Record<string, any>} [params] Optional query parameters to send with the API request.
+ */
+interface QueryConfig<TData, TError = AxiosError>
+  extends Omit<UseQueryOptions<TData, TError>, "queryKey" | "queryFn"> {
+  queryKey: any[];
+  endpoint: string;
+  params?: Record<string, any>;
+}
+
+/**
+ * A custom hook that fetches data from an API using the `useQuery` hook from `react-query`.
```

Comment:
I think this is an excellent example of how to write documentation, and it's worth us all following as a standard.